framework/src/st_things/things_stack/easy-setup/easysetup_manager.c

```diff
@@ -901,3 +906,59 @@ int esm_register_user_confirm_cb(user_confirm_result_func_type func)
 		return 0;
 	}
 }
+
+int esm_save_easysetup_state(int state)
+{
+	THINGS_LOG_D(THINGS_DEBUG, TAG, THINGS_FUNC_ENTRY);
+
+	if (state == 1) { // Done
+		THINGS_LOG_D(THINGS_DEBUG, TAG, "file open : %s", PATH_MNT FILE_ES_STATE);		
+		FILE *fp = fopen(PATH_MNT FILE_ES_STATE, "w+");
+		if (!fp) {
+			THINGS_LOG_D(THINGS_ERROR, TAG, "file open error(%d)", errno);
+			return -1;
+		}
+
+		unsigned char easysetup_completed = 1;
+		int ret = fwrite(&easysetup_completed, 1, sizeof(unsigned char), fp);
+		if (ret < 0) {
+			THINGS_LOG_D(THINGS_ERROR, TAG, "file write error(%d)", errno);
+			fclose(fp);
+			return -1;
+		}
+		fclose(fp);
+
+		THINGS_LOG_D(THINGS_DEBUG, TAG, "easysetup completed");
+	} else {
+		unlink(PATH_MNT FILE_ES_STATE);
+		THINGS_LOG_D(THINGS_DEBUG, TAG, "easysetup not completed");
+	}	
+
+	g_es_state = state;
+
+	return 0;
+}
+
+int esm_read_easysetup_state(void)
+{
+	THINGS_LOG_D(THINGS_DEBUG, TAG, "file open : %s", PATH_MNT FILE_ES_STATE);	
+	FILE *fp = fopen(PATH_MNT FILE_ES_STATE, "r");
+	if (!fp) {
+		THINGS_LOG_D(THINGS_ERROR, TAG, "file open error(%d)", errno);
+		return 0;
+	}
+
+	unsigned char easysetup_completed;
+	int ret = fread(&easysetup_completed, 1, sizeof(unsigned char), fp);
+	if (ret < 0) {
+		THINGS_LOG_D(THINGS_ERROR, TAG, "file write error(%d)", errno);
+		fclose(fp);
+		return 0;
```

Comment:
The caller should know the exact return value of fread(), not zero when fread fails.

Reply:
Thanks.